src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -409,25 +409,30 @@ private void errorHandler(RoutingContext routingContext) {
             routingContext.request().path());
 
         String message = null;
-        // in case of validation exception, building a meaningful error message
-        if (routingContext.failure() != null && routingContext.failure() instanceof ValidationException) {
-            ValidationException validationException = (ValidationException) routingContext.failure();
-            StringBuilder sb = new StringBuilder();
-            if (validationException.parameterName() != null) {
-                sb.append("Validation error on: " + validationException.parameterName() + " - ");
+        if (routingContext.statusCode() == HttpResponseStatus.BAD_REQUEST.code()) {
+            message = HttpResponseStatus.BAD_REQUEST.reasonPhrase();
+            // in case of validation exception, building a meaningful error message
+            if (routingContext.failure() != null && routingContext.failure() instanceof ValidationException) {
+                ValidationException validationException = (ValidationException) routingContext.failure();
+                StringBuilder sb = new StringBuilder();
+                if (validationException.parameterName() != null) {
+                    sb.append("Validation error on: " + validationException.parameterName() + " - ");
+                }
+                sb.append(validationException.getMessage());
+                message = sb.toString();
             }
-            sb.append(validationException.getMessage());
-            message = sb.toString();
+        } else if (routingContext.statusCode() == HttpResponseStatus.NOT_FOUND.code()) {
+            message = HttpResponseStatus.NOT_FOUND.reasonPhrase();
```

Comment:
Why do you only set the non-null message for NOT_FOUND and BAD_REQUEST?

Reply:
The error handler is set on the Vertx route only for these two errors. It means we are sure that it's called only in this two cases. All the others follow the normal flow.